gridstatus/ieso.py

```diff
@@ -844,3 +909,758 @@ def _request(self, url, verbose):
             )
 
         return r
+
+    def get_resource_adequacy_report(
+        self,
+        date: str | datetime.date | datetime.datetime,
+    ) -> pd.DataFrame:
+        """Retrieve and parse the Resource Adequacy Report for a given date.
+
+        Args:
+            date (str | datetime.date | datetime.datetime): The date for which to get the report
+
+        Returns:
+            pd.DataFrame: The Resource Adequacy Report df for the given date
+        """
+        json_data = self._get_resource_adequacy_json(date)
+        return self._parse_resource_adequacy_report(json_data)
+
+    # Note(Kladar): This might be fairly generalizable to other XML reports from IESO
+    def _get_resource_adequacy_json(
+        self,
+        date: str | datetime.date | datetime.datetime,
+    ) -> dict:
+        """Retrieve the Resource Adequacy Report for a given date and convert to JSON. There are often many
+        files for a given date, so this function will return the file with the highest version number. It does
```

Comment:
It seems likely we want to retrieve multiple versions of a report for a given date. I think we should support retrieving all versions in addition to the latest version.

Reply:
Okay, I've implemented that and will continue to test it out. 

It seems like the unversioned number (if it exists) also constitutes the latest of a date, but still confirming that and will update to handle that file as the latest. (It should be handled in the _all method now, but not yet the _latest method)

Reply:
Okay, I've seen enough to confirm that the unversioned file is also equivalent to the highest version number, and that it constitutes the `latest`
[ieso_versions_2024-09-08.csv](https://github.com/user-attachments/files/17849878/ieso_versions_2024-09-08.csv)
[ieso_versions_2024-09-09.csv](https://github.com/user-attachments/files/17849880/ieso_versions_2024-09-09.csv)
[ieso_versions_2024-12-01.csv](https://github.com/user-attachments/files/17849882/ieso_versions_2024-12-01.csv)

Reply:
Nice. That's what I saw as well.